[APP-3529] Update README with Chat API how-to (#86)
datarobot-oss/qa-app-streamlit: README.md

In this repository you will find the Q&amp;A Streamlit application code template used within DataRobot. The application uses a styling library for Streamlit called `streamlit-sal`, ([read more](https://github.com/datarobot-oss/streamlit-sal)).

> ## Deployment Configuration
> 
> The deployment used for this Q&A application needs to have an association ID configured and automatic association ID generation enabled.
> This setting can be found under Console -> Deployments -> `<Your deployment>` -> Settings -> Custom metrics
> 
> The `Association ID` name can be set to anything other than the reserved `promptText` or `resultText` values.

Reviewer: ```suggestion
## Deployment configuration

The deployment used for this Q&A application needs to have a configured association ID and automatic association ID generation enabled.

This setting can be found by navigating to **Console > Deployments > `<Your deployment>` > Settings > Custom metrics**.

The `Association ID` name can be set to anything other than the reserved `promptText` or `resultText` values.
```

./run_tests.sh
```

## Chat API

> LLM Blueprints created via DataRobots playground now support OpenAIs Chat API. The chat completion endpoint is made
> available via the deployment on: `<API_URL>/deployments/<deployment_id>/chat/completions`.
> Documentation for this API can be found here: https://platform.openai.com/docs/api-reference/chat

Reviewer: ```suggestion
LLM Blueprints created via DataRobot's Playground now support OpenAI's Chat API. The chat completion endpoint is made
available via the deployment on: `<API_URL>/deployments/<deployment_id>/chat/completions`.
Documentation for this API can be found [here](https://platform.openai.com/docs/api-reference/chat).
```

> Streaming will soon be supported by the above deployment Chat API implementation. As it is right now
> the streaming response always only contains 1 chunk due to the applied prompt and/or result text guards. 
> 
> Streaming can be disabled in the `constants.py` by setting `ENABLE_CHAT_API_STREAMING` to `False`
> 
> To disable the Chat API completely and continue to use the dataRobot-predict library, navigate to `constants.py`
> and set `FORCE_DISABLE_CHAT_API` to `True`

Reviewer: ```suggestion
Streaming will soon be supported by the deployment Chat API implementation described above. Currently, the streaming response always contains only one chunk due to the applied prompt and/or result text guards. 

Streaming can be disabled in the `constants.py` by setting `ENABLE_CHAT_API_STREAMING` to `False`.

To disable the Chat API completely and continue to use the dataRobot-predict library, navigate to `constants.py`
and set `FORCE_DISABLE_CHAT_API` to `True`.
```
